Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -526,99 +531,31 @@ private void EnsureMixedRealityRequirements()
-        {
-            get
-            {
-                if (IsInitialized)
-                {
-                    return instance;
-                }
-
-                if (Application.isPlaying && !searchForInstance)
-                {
-                    return null;
-                }
-
-
-                var objects = FindObjectsOfType<MixedRealityToolkit>();
-                searchForInstance = false;
-                MixedRealityToolkit newInstance;
-
-                switch (objects.Length)
-                {
-                    case 0:
-#if UNITY_EDITOR
-                        // Generating a new instance during app shutdown in the editor allows instances to persist into edit mode, which may cause unwanted behavior
-                        if (isApplicationQuitting)
-                        {
-                            Debug.LogWarning("Trying to find instance during application shutdown.");
-                            return null;
-                        }
-#endif
-                        Debug.Assert(!newInstanceBeingInitialized, "We shouldn't be initializing another MixedRealityToolkit unless we errored on the previous.");
-                        newInstanceBeingInitialized = true;
-                        newInstance = new GameObject(nameof(MixedRealityToolkit)).AddComponent<MixedRealityToolkit>();
-                        break;
-                    case 1:
-                        newInstanceBeingInitialized = false;
-                        newInstance = objects[0];
-                        break;
-                    default:
-                        newInstanceBeingInitialized = false;
-                        Debug.LogError($"Expected exactly 1 {nameof(MixedRealityToolkit)} but found {objects.Length}.");
-                        return null;
-                }
-
-                Debug.Assert(newInstance != null);
-
-                if (!isApplicationQuitting)
-                {
-                    // Setup any additional things the instance needs.
-                    newInstance.InitializeInstance();
-                }
-                else
-                {
-                    // Don't do any additional setup because the app is quitting.
-                    instance = newInstance;
-                    newInstanceBeingInitialized = false;
-                }
-
-                Debug.Assert(instance != null);
-
-                return instance;
-            }
-        }
+        private static readonly object initializedLock = new object();
 
         /// <summary>
-        /// Lock property for the Mixed Reality Toolkit to prevent reinitialization
+        /// Returns the Singleton instance of the classes type.
         /// </summary>
-        private static readonly object initializedLock = new object();
+        public static MixedRealityToolkit Instance => activeInstance;
 
         private void InitializeInstance()
         {
-            lock (initializedLock)
```

Comment:
do we no longer need this lock?

Reply:
It seems to me we could remove it, but I didn't want to take that step until someone else had looked at it.

Reply:
Well the instance is a monobehavior, and those fail when being called from other threads, so the lock is entirely superfluous

Reply:
Well the instance is a monobehavior, and those fail when being called from other threads, so the lock is entirely superfluous

Reply:
Well the instance is a monobehavior, and those fail when being called from other threads, so the lock is entirely superfluous